### PR TITLE
fix(yargs): add error parameter to getCompletion callback

### PR DESCRIPTION
**Pull request by richardkazuomiller**
The type of the callback for `getCompletion` has changed since v17 and the type declarations are outdated.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/yargs/yargs/commit/6ad2be0cb564cbcf01f5148db05e8cf999057caf#diff-a6b47d9a7a59d2fffb8a85bce2005db20344af8fc1b70a8dddd6400633ba1862>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/yargs/ts4.1/index.d.ts [hunk 1]
```diff
@@ -325,7 +325,7 @@ declare namespace yargs {
          * @param args An array of the words in the command line to complete.
          * @param done The callback to be called with the resulting completions.
          */
-        getCompletion(args: ReadonlyArray<string>, done: (completions: ReadonlyArray<string>) => void): Argv<T>;
+         getCompletion(args: ReadonlyArray<string>, done: (err: Error|null, completions: ReadonlyArray<string>) => void): Argv<T>;
```
Comment (forivall) [inline types/yargs/ts4.1/index.d.ts:328]: Should also include a promise signature ([source code](https://github.com/yargs/yargs/commit/6ad2be0cb564cbcf01f5148db05e8cf999057caf#diff-452e4f3371a75b323f89da8de1a1707b1532abef67be1a02fe7e8396e0f35020R1385)):
```suggestion
        getCompletion(args: ReadonlyArray<string>, done: (err: Error|null, completions: ReadonlyArray<string>) => void): Argv<T>;
        getCompletion(args: ReadonlyArray<string>, done?: never): Promise<ReadonlyArray<string>>;
        
```
(and add tests for the above suggestion)

**Review by forivall — CHANGES_REQUESTED** [review]

File: types/yargs/ts4.1/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardkazuomiller) [reply types/yargs/ts4.1/index.d.ts]: Done! 🤠

Comment (richardkazuomiller) [reply types/yargs/ts4.1/index.d.ts]: Done! Thanks for catching that 🤠

**Review by forivall — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**